### PR TITLE
AST visitor for value translator

### PR DESCRIPTION
**Pull request by brodyf**

### CODE REVIEW

File: src/AstTranslator.cpp [hunk 1]
```diff
@@ -225,55 +225,80 @@ std::unique_ptr<RamRelation> AstTranslator::translateRelation(
 }
 
 std::unique_ptr<RamValue> AstTranslator::translateValue(const AstArgument* arg, const ValueIndex& index) {
-    std::unique_ptr<RamValue> val;
-    if (!arg) {
-        return val;
-    }
+    class ValueTranslator : public AstVisitor<std::unique_ptr<RamValue>> {
```
Comment (mmcgr) [inline src/AstTranslator.cpp]: The check for nullptr is gone now.
```suggestion
    if (arg == nullptr) {
        return std::unique_ptr<RamValue>();
    }
    class ValueTranslator : public AstVisitor<std::unique_ptr<RamValue>> {
```

Comment (b-scholz) [reply src/AstTranslator.cpp]: Do we know when a nullptr occurs? That feels not right.

Comment (mmcgr) [reply src/AstTranslator.cpp]: The failed tests were all related to aggregation, I think
semantic/aggregates2
semantic/bool
evaluation/aggregates
evaluation/average
evaluation/count_sccs1
evaluation/magic_aggregates
evaluation/magic_dominance

When AstTranslator reaches line 546  the AstAggregator has a null target expression.
```
// translate target expression
std::unique_ptr<RamValue> value = translateValue(cur->getTargetExpression(), valueIndex);
```
So, anything with a null target expression.

Comment (mmcgr) [reply src/AstTranslator.cpp]: I think that's anything that uses count

**Pull request merged**